server/routerlicious/packages/lambdas/src/scriptorium/lambda.ts

```diff
@@ -69,6 +70,34 @@ export class ScriptoriumLambda implements IPartitionLambda {
 		}
 
 		this.pendingOffset = message;
+
+		if (this.telemetryEnabled) {
+			if (this.pendingMetric === undefined) {
+				// create a new metric for processing the current kafka batch
+				this.pendingMetric = Lumberjack.newLumberMetric(
+					LumberEventName.ScriptoriumProcessBatch,
+					{
+						timestampReadyToProcess: Date.now(),
+						kafkaPartition: this.pendingOffset?.partition,
+						batchOffsetStart: this.pendingOffset?.offset,
+						batchOffsetEnd: this.pendingOffset?.offset,
```

Comment:
Quick q: batchOffsetStart and batchOffsetEnd seem to be the same? If so, do we need both?

Reply:
batchOffsetEnd will be overwirtten in the else block below, i.e. if multiple offsets are being processed in the same scriptorium batch. In other words, this is to log the range of offsets being processed (start and end), and yes they will be same in case of single offset.